TFNetworkRecLayer.py

```diff
@@ -2214,9 +2268,10 @@ def cond(i, net_vars, acc_tas, seq_len_info=None):
       [tf.TensorShape(None) for _ in init_acc_tas])
     if not have_known_seq_len:
       # See body().
-      out_batch_dim = self.layer_data_templates["end"].get_batch_dim()
+      end_batch_dim = self.layer_data_templates["end"].get_batch_dim()
+      out_batch_dim = self.layer_data_templates["output"].get_batch_dim()
       init_seq_len_info = (
-        constant_with_shape(False, shape=[out_batch_dim], name="initial_end_flag"),
+        constant_with_shape(False, shape=[end_batch_dim], name="initial_end_flag"),
         constant_with_shape(0, shape=[out_batch_dim], name="initial_seq_len"))
```

Comment:
I don't think it is a good idea to mix it. As said, the end layer should just be resolved automatically to the latest choice. And the same for the "output" layer.